File: solution/661. Image Smoother/661.cpp
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
        const int m = img.size(), n = img[0].size();
        vector<vector<int>> res(m, vector<int>(n));
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                int cnt = 0, sum = 0;
                for (int y = max(0, i - 1); y < min(m, i + 2); ++y) {
                    for (int x = max(0, j - 1); x < min(n, j + 2); ++x) {
                        cnt += 1;
                        sum += img[y][x];
                    }
                }
                res[i][j] = sum / cnt;
            }
        }
        return res;
    }
};

File: solution/661. Image Smoother/661.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m, n = len(img), len(img[0])
        res = [[0] * n for i in range(m)]
        for i in range(m):
            for j in range(n):
                s, cnt = 0, 0
                for y in range(max(0, i-1), min(m, i+2)):
                    for x in range(max(0, j-1), min(n, j+2)):
                        s += img[y][x]
                        cnt += 1
                res[i][j] = s // cnt;
        return res
    
